roles/validations/tasks/edpm/hugepages_and_reboot.yml

```diff
@@ -0,0 +1,121 @@
+  register: deployed_nodeset_name
+
+  # collect initial confighash from the nodeset
+- name: Collecting initial confighash from the nodeset
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell:
+    cmd: >-
+      set -o pipefail && oc get -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }} osdpns "{{ deployed_nodeset_name.stdout }}" -o yaml | yq .status.configHash
+  register: initial_config_hash
+
+- name: Update hugepages value
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.command:
+    cmd: >-
+      oc patch -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }} osdpns/"{{ deployed_nodeset_name.stdout }}" --type=merge -p '{"spec": {"nodeTemplate": {"ansible": {"ansibleVars": {"edpm_kernel_args": "default_hugepagesz=1gb hugepagesz=1g hugepages=64 iommu=pt intel_iommu=on tsx=off isolcpus=2-11,14-23"}}}}}'
+
+# The reboot-os service doesn't exist by default. So we need to create this service in
+# order to reboot the nodes.
+- name: Create reboot service
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell:
+    cmd: |
+      oc apply -f - <<EOF
+      apiVersion: dataplane.openstack.org/v1beta1
+      kind: OpenStackDataPlaneService
+      metadata:
+        name: reboot-os
+      spec:
+        playbook: osp.edpm.reboot
+      EOF
+
+# Create a new OpenStackDataPlaneDeployment to set the reboot variable. We also only need to execute
+# the jobs required to update Hugepages and reboot the nodes.
+- name: Create openstackdataplanedeployment to rollout changes
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell:
+    cmd: |
+      oc apply -f - <<EOF
+      apiVersion: dataplane.openstack.org/v1beta1
+      kind: OpenStackDataPlaneDeployment
+      metadata:
+        name: edpm-hugepages-update
+        namespace: {{ cifmw_install_yamls_defaults['NAMESPACE'] }}
+      spec:
+        nodeSets:
+          - "{{ deployed_nodeset_name.stdout }}"
+        ansibleExtraVars:
+          edpm_reboot_force_reboot: "true"
+        servicesOverride:
+          - bootstrap
+          - reboot-os
+      EOF
+
+# loop check the status of the openstackdataplanedeployment until it is either completed,
+# or reaches a defined timeout value.
+- name: Wait for deployment to be complete
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.command:
+    cmd: >-
+      oc wait openstackdataplanedeployment edpm-hugepages-update
+      --namespace={{ cifmw_install_yamls_defaults['NAMESPACE'] }}
+      --for=condition=ready
+      --timeout={{ cifmw_validations_timeout | default(100) }}m
+
+# collect initial confighash from the nodeset
+- name: Collecting initial confighash from the nodeset
```

Comment:
nit: this is not the initial one

Reply:
Dropped this commit from the PR since it has it's own PR open and I have indeed fixed that over there:
https://github.com/openstack-k8s-operators/ci-framework/pull/1356/files#diff-87bc850ca533dc975466769e9c72824ab77a1c6a9f6e69856277fd63be9f9e7bR99